R/plot_plate.R

```diff
@@ -0,0 +1,273 @@
+  )
+
+  # Add colors to the well positions data frame
+  well_positions$color <- colors
+  well_positions$numbers <- numbers
+
+  # Define the aspect ratio of the background image
+  background_image_resolution <- c(dim(plate_img)[2], dim(plate_img)[1])
+  aspect_ratio <- background_image_resolution[2] / background_image_resolution[1]
+  area_size <- min(dev.size("px") / background_image_resolution)
+
+
+  categories <- names(legend_mapping)
+  well_positions$category <- factor(well_positions$color, levels = legend_mapping, labels = categories)
+
+
+  # Plot the plate with colored wells
+  p <- ggplot(well_positions, aes(x = x, y = y)) +
+    annotation_custom(
+      rasterGrob(rgb_image, width = unit(1, "npc"), height = unit(1, "npc")),
+      -Inf, Inf, -Inf, Inf
+    ) +
+    geom_point(aes(fill = category), size = area_size * 19 - 1, shape = 21, color = "black", stroke = 0) +
+    scale_fill_manual(values = legend_mapping) +
+    theme_void() +
+    scale_x_continuous(limits = c(0, 1)) +
+    scale_y_continuous(limits = c(0, 1)) +
+    ggtitle(plot_title) +
+    theme(
+      aspect.ratio = aspect_ratio,
+      plot.title = element_text(hjust = 0.5, size = area_size * 20 + 5, vjust = -1),
+      legend.title = element_text(size = 0),
+      legend.text = element_text(size = 12, face = "bold"),
+      legend.background = element_rect(fill = "white", linewidth = 0),
+      legend.key = element_rect(fill = "white", color = "white")
+    )
+
+  if ((dev.size("px") / background_image_resolution)[1] < (dev.size("px") / background_image_resolution)[2]){
+    p <- p + theme(legend.position = "bottom")
+  }
+
+  if (plot_numbers) {
+    p <- p + geom_text(aes(label = numbers), size = area_size * 8 - 0.5, color = "black", vjust = 0.5, hjust = 0.5, fontface = "bold")
+  }
+
+  if (!plot_legend) {
+    p <- p + theme(legend.position = "none")
+  }
+
+  p
+}
+
+
+#' Plot counts in a 96-well plate
+#'
+#' It is a function to plot counts in a 96-well plate using a color scale.
+#' The function uses a color scale to map the counts to colors and plot them
+#' in a 96-well plate.
+#' The function uses the plot_plate function to plot the counts.
+#'
+#' @param plate The plate object with the counts data
+#' @param analyte_name The name of the analyte
+#' @param plot_counts Logical indicating if the counts should be plotted
+#' @param plot_legend Logical indicating if the legend should be plotted
+#'
+#' @return A ggplot object
+#'
+#' @examples
+#' plate_filepath <- system.file("extdata", "CovidOISExPONTENT_CO.csv", package = "PvSTATEM", mustWork = TRUE)
+#' layout_filepath <- system.file("extdata", "CovidOISExPONTENT_CO_layout.xlsx", package = "PvSTATEM", mustWork = TRUE)
+#' plate <- read_luminex_data(plate_filepath, layout_filepath)
+#' plot_counts(plate = plate, analyte_name = "OC43_NP_NA", plot_counts = T, plot_legend = F)
+#'
+#' @export
+plot_counts <- function(plate, analyte_name = NULL, plot_counts = FALSE, plot_legend = FALSE) {
```

Comment:
Are `plot_counts = FALSE` and `plot_legend = FALSE` sane defaults? I suppose that this function will be called by the user (because it is exported) then as a user I would not like to have to type TRUE into two arguments each time. Also I would set `analyte_name` as a positional argument. (In R the positional arguments can be also provided in a keyword argument style)

Reply:
I think this plot is self explanatory, and doesn't need legend by deafult: green - good, red - something wrong. Numbers add additional clutter and I suspect user won't examine it in such great detail. This plot serves purpose as a quick indicator if everything is alright. Thus I don't think `plot_counts = TRUE` by deafult is justified. 

analyse_name should be a positional argument

Reply:
"Thus I don't think plot_counts = TRUE by deafult is justified."
I understand your point of view, but I don't fully agree with the overcluttering argument. I think that it will be best to ask the actual users what they think